express-zod-api/src/zts.ts

```diff
@@ -41,47 +61,65 @@ const nodePath = {
   optional: R.path(["questionToken" satisfies keyof ts.TypeElement]),
 };
 
-const onLiteral: Producer = ({ value }: z.ZodLiteral<LiteralType>) =>
-  makeLiteralType(value);
+const onLiteral: Producer = ({ _zod: { def } }: $ZodLiteral) => {
+  const values = def.values.map((entry) =>
+    entry === undefined
+      ? ensureTypeNode(ts.SyntaxKind.UndefinedKeyword)
+      : makeLiteralType(entry),
+  );
+  return values.length === 1 ? values[0] : f.createUnionTypeNode(values);
+};
+
+const onInterface: Producer = (int: $ZodInterface, { next, makeAlias }) =>
```

Comment:
this would be required to combine with `onObject` when the following PR merged and released:
https://github.com/colinhacks/zod/pull/4271